### PR TITLE
Optimize message size calculation

### PR DESCRIPTION
**Pull request by sagarkrkv**
Fix issue #328

### CODE REVIEW

File: message.go [hunk 1]
```diff
@@ -37,6 +37,11 @@ func (msg Message) message(cw *crc32Writer) message {
 	}
 	return m
 }
+func (msg Message) size() int32 {
+	size := 4 + 1 + 1 + sizeofBytes(msg.Key) + sizeofBytes(msg.Value)
+	size += 8 // Timestamp
```
[the remark this answers is not in the document]
Comment (sagarkrkv) [reply message.go]: I wanted to preserve the comment that says that 8 is the size of the timestamp.

**Comment by ShaneSaww** [comment]
@sagarkrkv Do you have the results of the benchmark from https://github.com/segmentio/kafka-go/pull/329/commits/d7b290efa55161eba3169817c7677a88c2c304be

I would be interested in seeing the before and after.

**Comment by sagarkrkv** [comment]
I removed that benchmark as there was a lot of noise that prevented me from getting reliable measurements.

```go
BenchmarkMessageSize/0.2.5-8            5000000             296 ns/op          0 B/op          0 allocs/op
BenchmarkMessageSize/0.3.1-8         	50000000	        25.6 ns/op	       0 B/op	       0 allocs/op
BenchmarkMessageSize/with_this_pr-8     300000000	        4.03 ns/op	       0 B/op	       0 allocs/op

**Comment by sagarkrkv** [comment]
@achille-roussel do you have anymore comments on this PR?

**Pull request merged**